R/source.R

```diff
@@ -248,23 +248,28 @@ invoke_cargo <- function(toolchain, specific_target, dir, profile,
       )
     }
 
-    # rtools_path() returns path to the RTOOLS40_HOME\usr\bin,
-    # but we need RTOOLS40_HOME\mingw{arch}\bin, hence the "../.."
-    rtools_home <- normalizePath(
-      # `pkgbuild` may return two paths for R < 4.2 if Rtools42 is present
-      file.path(pkgbuild::rtools_path()[1], "..", ".."),
-      winslash = "/",
-      mustWork = TRUE
-    )
-
-    rtools_bin_path <-
-      normalizePath(
-        file.path(
-          rtools_home,
-          paste0("mingw", ifelse(R.version$arch == "i386", "32", "64")),
-          "bin"
-        )
+    if (identical(R.version$crt, "ucrt")) {
+      # TODO: rtools_path() currently returns RTOOLS42_HOME for R >= 4.2, but
+      # current R-devel still uses Rtools40 (c.f. https://community.rstudio.com/t/what-is-or-can-be-the-plan-to-support-rtools42-on-r-lib-actions/125114/4).
+      # So, we need a tweak for now. Eventually, this can be removed.
+      rtools_home <- Sys.getenv("RTOOLS40_HOME", "C:\\rtools42")
```

Comment:
Out of curiosity, why can't we use here `pkgbuild::rtools_path()`? Their latest release has some support for two versions of Rtools on the system.

Reply:
`pkgbuild::rtools_path()` always returns the path to Rtools42, which isn't used yet.

https://github.com/r-lib/pkgbuild/blob/161f7d31d23da1c89fe9959679a6fd3150eba515/R/rtools.R#L34

> Their latest release has some support for two versions of Rtools on the system.

It's a bit confusing, but what they added was a support for Rtools40v2, which contains both toolchain for R 4.1 (`/mingw*`) and that for R 4.2 (`/ucrt64`).

Reply:
Ok, I see.